diff --git a/gcc-4.9/gcc/config/i386/linux-common.h b/gcc-4.9/gcc/config/i386/linux-common.h
index 51e25b4..7f70bd4 100644
--- a/gcc-4.9/gcc/config/i386/linux-common.h
+++ b/gcc-4.9/gcc/config/i386/linux-common.h
@@ -32,7 +32,7 @@ along with GCC; see the file COPYING3.  If not see
   LINUX_OR_ANDROID_CC (GNU_USER_TARGET_CC1_SPEC, \
                        GNU_USER_TARGET_CC1_SPEC \
                        " -mstackrealign -msse3" \
-                       " -m32 -fno-short-enums" \
+                       " -fno-short-enums" \
                        " " \
                        ANDROID_CC1_SPEC("-fPIC"))
 
-- 
1.8.1.4

