diff --git a/build/tools/dev-defaults.sh b/build/tools/dev-defaults.sh
index d569ab4..b612034 100644
--- a/build/tools/dev-defaults.sh
+++ b/build/tools/dev-defaults.sh
@@ -29,7 +29,7 @@ LIBPORTABLE_SUBDIR=sources/android/libportable
 TOOLCHAIN_GIT_DATE=now
 
 # The space-separated list of all GCC versions we support in this NDK
-DEFAULT_GCC_VERSION_LIST="4.6 4.7 4.8 4.4.3"
+DEFAULT_GCC_VERSION_LIST="4.4.3 4.6 4.7 4.8 4.9"
 
 # The default GCC version for this NDK, i.e. the first item in
 # $DEFAULT_GCC_VERSION_LIST
diff --git a/build/tools/dev-rebuild-ndk.sh b/build/tools/dev-rebuild-ndk.sh
index 4f6b424..ade52ee 100755
--- a/build/tools/dev-rebuild-ndk.sh
+++ b/build/tools/dev-rebuild-ndk.sh
@@ -65,7 +65,7 @@ linux )
     if [ "$OPTION_QUICK_BUILD" = "yes" ]; then
        BUILD_TARGET_PLATFORMS="linux-x86"
     else
-       BUILD_TARGET_PLATFORMS="linux-x86 windows"
+       BUILD_TARGET_PLATFORMS="linux-x86"
     fi
     ;;
 darwin )
@@ -114,7 +114,9 @@ fi
 
 echo
 echo "Checking for Toolchain sources"
-NDK_SRC_DIR=/tmp/ndk-$USER/src/android-ndk-src-${TOOLCHAIN_GIT_DATE}${PATCHES_SHA1}
+#NDK_SRC_DIR=/tmp/ndk-$USER/src/android-ndk-src-${TOOLCHAIN_GIT_DATE}${PATCHES_SHA1}
+NDK_SRC_DIR=%rootDir%/repositories/toolchain_synced
+
 if [ ! -d $NDK_SRC_DIR ]
 then
     echo "  Downloading Toolchain sources"
