project external/grub/
diff --git a/Android.mk b/Android.mk
index 50034ba..3f57042 100644
--- a/Android.mk
+++ b/Android.mk
@@ -46,7 +46,7 @@ include $(BUILD_RAW_EXECUTABLE)
 $(LOCAL_BUILT_MODULE) : PRIVATE_LINK_SCRIPT :=
 $(LOCAL_BUILT_MODULE) : PRIVATE_LIBS :=
 $(LOCAL_BUILT_MODULE) : PRIVATE_RAW_EXECUTABLE_LDFLAGS := \
-	-nostdlib -N -Ttext=7C00 -melf_i386
+	-nostdlib -N -Ttext=0x7C00 -melf_i386
 
 ###################################################################
 ###################################################################
@@ -109,7 +109,7 @@ include $(BUILD_RAW_EXECUTABLE)
 $(LOCAL_BUILT_MODULE) : PRIVATE_LINK_SCRIPT :=
 $(LOCAL_BUILT_MODULE) : PRIVATE_LIBS :=
 $(LOCAL_BUILT_MODULE) : PRIVATE_RAW_EXECUTABLE_LDFLAGS := \
-	-nostdlib -N -Ttext=8200 -melf_i386
+	-nostdlib -N -Ttext=0x8200 -melf_i386
 
 #############################################
 ## Generate the stage2 start file  (2) + (3)
@@ -159,7 +159,7 @@ $(all_objects): $(STAGE2_SIZE_OBJ)
 $(LOCAL_BUILT_MODULE) : PRIVATE_LINK_SCRIPT :=
 $(LOCAL_BUILT_MODULE) : PRIVATE_LIBS :=
 $(LOCAL_BUILT_MODULE) : PRIVATE_RAW_EXECUTABLE_LDFLAGS := \
-	-nostdlib -N -Ttext=8200 -melf_i386
+	-nostdlib -N -Ttext=0x8200 -melf_i386
 
 #############################################
 ## Generate the real deal stage2  (4)

