--- types.h	2012-06-14 13:05:48.610501000 +0400
+++ types.h	2012-06-14 16:43:47.000000000 +0400
@@ -39,6 +39,21 @@
 #include <linux/types.h>
 #include <machine/kernel.h>
 
+
+
+#if !defined(__cplusplus) || defined(__STDC_LIMIT_MACROS)
+#  define __STDINT_LIMITS
+#endif
+
+#if !defined(__cplusplus) || defined(__STDC_CONSTANT_MACROS)
+#  define  __STDINT_MACROS
+#endif
+
+#if !defined __STRICT_ANSI__ || __STDC_VERSION__ >= 199901L
+#  define __STDC_INT64__
+#endif
+
+
 typedef __u32    __kernel_dev_t;
 
 /* be careful with __kernel_gid_t and __kernel_uid_t
@@ -121,7 +136,9 @@
 typedef uint32_t       u_int32_t;
 typedef uint16_t       u_int16_t;
 typedef uint8_t        u_int8_t;
+#if defined(__STDC_INT64__)
 typedef uint64_t       u_int64_t;
 #endif
+#endif
 
 #endif
